crates/cast/src/helpers/build.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{Context, Result};
+use scarb_artifacts::{get_contracts_map, target_dir_for_package, StarknetContractArtifacts};
+use scarb_metadata::{Metadata, PackageMetadata};
+use scarb_ui::args::PackagesFilter;
+use std::collections::HashMap;
+use std::fs;
+use std::process::{Command, Stdio};
+
+use super::constants::LIB_CONTRACT_ARTIFACTS_NAME;
+
+#[allow(clippy::too_many_lines)]
+pub fn build(
```

Comment:
It not only builds but also loads them to memory, please split it into two separate functions.

Reply:
I'm not entirely sure about that. I think it makes sense for a single function to return the artifacts that it built

Reply:
I guess we could extract scripts part to separate function and only call it in scripts, that would be probably more optimal